verifiers/interfaceValidator.js

```diff
@@ -0,0 +1,85 @@
+"use strict";
+
+const PLUGIN_NAME = 'Interface Validator';
+var utils = require('./utilities');
+var path = require('path');
+var glob = require("glob");
+var raml = require('raml-1-parser');
+var acorn = require('acorn/dist/acorn_loose');
+acorn.walk = require('acorn/dist/walk');
+
+// Interface Validator ensures that thing translator classes have implementations
+// for all methods defined in the raml for the device type.
+function verify(file, log) {
+  var ramlPath = getRamlFilePath(file.path);
+  var schemaMethods = getSchemaMethods(ramlPath);
+  var translatorMethods = getTranslatorMethods(file);
+
+  schemaMethods.forEach(method => {
+    if(translatorMethods.indexOf(method) === -1) {
+      log.error(file.path, 'Schema method not found: ' + method);
+    }
+  });
+}
+
+// Get the raml file that corresponds to the thing translator.
+// This assumes that the nearest ancestor raml file is the correct
+// corresponding file.
+function getRamlFilePath(filePath) {
+  var prevDir = null;
+  var currentDir = path.dirname(filePath);
+
+  while(prevDir !== currentDir) {
+    var ramlFiles = glob.sync(currentDir + '/*.raml');
+
+    if(ramlFiles.length > 0) {
+      return ramlFiles[0];
+    }
+
+    prevDir = currentDir;
+    currentDir = path.join(currentDir, '..');
```

Comment:
Yes, this will stop looking after it reaches the root directory.  If no raml file is found by then it assumes there are no schema methods to look for.

---
In reply to: [87662212](https://github.com/openT2T/translators/pull/119#discussion_r87662212) [](ancestors = 87662212,87661150)